### PR TITLE
control-service: Allow for jobs with no schedule to be deployed

### PR DESCRIPTION
**Pull request by gageorgiev**
Currently, if a job is deployed with no schedule_cron,
the deployment will fail. After this change, jobs with
no schedule will be deployed successfully, their cronjobs
will have a default schedule "0 0 30 2 *" - meaning Feb 30,
or never - but the job metadata will have an empty string
for a schedule, so that "vdk list" will not print a schedule
for those jobs (but jobs with their schedule explicitly set
to "0 0 30 2 *" will still have it show).

Testing done: tested locally by running control-service
connected to deployed K8s cluster, created and deployed
job successfully, verified that the created cronjob object
has its schedule set to "0 0 30 2 *", that executing the job
through the Execution API works and that getting the list of
deployed jobs returns an empty string for the schedule param

included 1 unit test covering the new addition

Signed-off-by: Gabriel Georgiev <gageorgiev@vmware.com>

### CODE REVIEW

**Comment by mivanov1988** [comment]
LGTM! Please take a look at the failed tests.

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/JobImageDeployerTest.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public void testScheduleLabelsAnnotations() {
             var annotations = annotationCaptor.getValue();
             //check everything was as expected
             Assertions.assertEquals(3, labels.size(), "Expecting three labels");
-            Assertions.assertEquals(5, annotations.size(), "Expecting five annotations");
+            Assertions.assertEquals(6, annotations.size(), "Expecting five annotations");
```
Comment (doks5) [inline projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/JobImageDeployerTest.java:101]: Do we still expect five annotations?

**Review by doks5 — APPROVED** [review]
Overall, LGTM

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/JobImageDeployerTest.java (hunk 1 above)
Comment (gageorgiev) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/JobImageDeployerTest.java:101]: good catch

**Review by mivanov1988 — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/JobImageDeployerTest.java [hunk 2]
```diff
@@ -113,13 +113,15 @@ public void testScheduleLabelsAnnotations() {
...
             var executionType = annotations.get(JobAnnotation.EXECUTION_TYPE.getValue());
+            var unscheduled = annotations.get(JobAnnotation.UNSCHEDULED.getValue());
 
             Assertions.assertEquals("schedule string", schedule);
             Assertions.assertEquals("lastDeployedBy", deployedBy);
             Assertions.assertEquals(OffsetDateTime.now().toString().substring(0, 10), deployedDate.toString().substring(0, 10),
                             "Testing if date is the same, we cannot be precise down to the millisecond.");
             Assertions.assertEquals("scheduled/runtime", startedBy);
             Assertions.assertEquals("scheduled", executionType);
+            Assertions.assertEquals("false", unscheduled);
```
Comment (ivakoleva) [inline projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/JobImageDeployerTest.java:124]: All k8s jobs are currently "com.vmware.taurus/execution-type: scheduled" annotated, correct? 
What are the annotation changes expected with this PR, maybe add to description?

**Review by ivakoleva — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/JobImageDeployerTest.java (hunk 2 above)
Comment (gageorgiev) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/JobImageDeployerTest.java:124]: Jobs will remain annotated `execution-type: scheduled` as per Miro's comment higher up.

**Pull request merged**